luigi/contrib/pig.py

```diff
@@ -95,27 +95,29 @@ def pig_script_path(self):
         """
         raise NotImplementedError("subclass should define pig_script_path")
 
+    @contextmanager
```

Comment:
Oh I see, I didn't think that this kind of complication would arise as of this. Oh well, it had to be done.